misc/synchronization.md

```diff
@@ -36,6 +36,7 @@ however they cannot be compiled since they are simplified to improve readability
 ```cpp
   Matrix m;
 
+  // TODO: These should probably be updated to use sender algorithms?
   pika::dataflow(Task1, m({0, 0}));
```

Comment:
What would you like to do about the documentation? Eventually this needs be updated.

Reply:
I think we can add a note on outdated docs, and wait when the sender/receiver port is completed to update it.

Reply:
https://github.com/eth-cscs/DLA-Future/issues/476